src/components/TextField/TextField.tsx

```diff
@@ -52,6 +53,8 @@ export interface BaseProps {
   labelHidden?: boolean;
   /** Disable the input */
   disabled?: boolean;
+  /** Show a clear text button in the input */
+  clearButton?: boolean;
```

Comment:
Rather than add an additional prop should we only show the clear button if `type="search"`?

Reply:
Good question @AndrewMusgrave . The [issue](https://github.com/Shopify/polaris-react/issues/1115) suggested adding a prop, but I'm not sure if there is a use case for a clear button on non search inputs.

I wonder if we should only have this on search inputs, but still have the prop so that it could be set to `false` in the event that a clear button is not wanted on a search input?

Reply:
I agree with your approach here @andrewmcgov. There may be cases where we want to use the clear button even when the type is not search.

Let's keep it the way you have it, and have it not dependent on the input type.